storage/mediator.go

```diff
@@ -199,3 +219,33 @@ func (m *mediator) reportLoop() {
 		}
 	}
 }
+
+func (m *mediator) databaseBootstrapStates() (databaseBootstrapStates, error) {
+	namespaces, err := m.database.GetOwnedNamespaces()
+	if err != nil {
+		return databaseBootstrapStates{}, err
+	}
+
+	states := databaseBootstrapStates{
+		namespaceBootstrapStates: make(map[string]shardBootstrapStates, len(namespaces)),
+	}
+
+	for _, namespace := range namespaces {
+		nsState := make(shardBootstrapStates)
+
+		for _, shard := range namespace.GetOwnedShards() {
+			nsState[shard.ID()] = shard.IsBootstrapped()
+		}
+		states.namespaceBootstrapStates[namespace.ID().String()] = nsState
+	}
+
+	return states, nil
+}
+
+// databaseBootstrapStates storse a snapshot of the bootstrap state for all shards across all
+// namespaces at a given moment in time.
+type databaseBootstrapStates struct {
```

Comment:
nit: rename to `databaseBootstrapState`, also could you move the decls to `storage/types.go`

Reply:
done